src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -1222,486 +1631,614 @@ s32 func_80A13B1C(EnGo* this, PlayState* play) {
+ * sitting, and helping the hamstring stretching goron.)
+ */
+void EnGo_Idle(EnGo* this, PlayState* play) {
+    s16 targetRot = this->actor.world.rot.y;
 
-    if ((ENGO_GET_F(&this->actor) == ENGO_F_2) && (gSaveContext.save.entrance == ENTRANCE(GORON_RACETRACK, 1))) {
+    if ((ENGO_GET_TYPE(&this->actor) == ENGO_SPECTATOR) &&
+        (gSaveContext.save.entrance == ENTRANCE(GORON_RACETRACK, 1))) {
+        // Spectators only cheer. No other interactions
         Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_GORON_CHEER - SFX_FLAG);
-    } else if (ENGO_GET_F(&this->actor) != ENGO_F_8) {
-        if (func_80A1222C(this, play)) {
-            SubS_UpdateFlags(&this->unk_390, 0, 7);
-            this->unk_3EC = 1;
-            this->actionFunc = func_80A14B30;
-        } else if (ENGO_GET_F(&this->actor) == ENGO_F_4) {
-            switch (ENGO_GET_70(&this->actor)) {
-                case ENGO_70_0:
-                    func_80A132C8(this, play);
+    } else if (ENGO_GET_TYPE(&this->actor) != ENGO_PKEG_SELLER) {
+        // All others besides the Powder Keg Seller can be slept,
+        if (EnGo_IsEnteringSleep(this, play)) {
+            SubS_UpdateFlags(&this->actionFlags, 0, 7);
+            this->sleepState = ENGO_ASLEEP_POS;
+            this->actionFunc = EnGo_Sleep;
+        } else if (ENGO_GET_TYPE(&this->actor) == ENGO_GRAVEBRO) {
+            switch (ENGO_GET_SUBTYPE(&this->actor)) {
+                case ENGO_GRAVEBRO_GRAVEMAKER:
+                    EnGo_GraveBroGoron_Gravemaker_Idle(this, play);
                     break;
 
-                case ENGO_70_1:
-                    func_80A133A8(this, play);
+                case ENGO_GRAVEBRO_FROZEN:
+                    EnGo_GraveBroGoron_Frozen_Idle(this, play);
                     break;
             }
-        } else if (ENGO_GET_F(&this->actor) == ENGO_F_1) {
+        } else if (ENGO_GET_TYPE(&this->actor) == ENGO_STRETCHER) {
             if (ABS_ALT(BINANG_SUB(this->actor.yawTowardsPlayer, this->actor.shape.rot.y)) < 0x3FFC) {
-                SubS_UpdateFlags(&this->unk_390, 3, 7);
+                SubS_UpdateFlags(&this->actionFlags, 3, 7);
             } else {
-                SubS_UpdateFlags(&this->unk_390, 0, 7);
+                SubS_UpdateFlags(&this->actionFlags, 0, 7);
             }
         }
     }
-    Math_ApproachS(&this->actor.shape.rot.y, sp26, 4, 0x2AA8);
+    Math_ApproachS(&this->actor.shape.rot.y, targetRot, 4, 0x2AA8);
 }
 
-void func_80A14B30(EnGo* this, PlayState* play) {
-    s16 sp26 = this->actor.world.rot.y;
+/**
+ * Handles the entry into sleep, as well as snoring while asleep
+ */
+void EnGo_Sleep(EnGo* this, PlayState* play) {
+    s16 targetRot = this->actor.world.rot.y;
     u16 sfxId;
 
-    if (func_80A1222C(this, play)) {
-        this->unk_3EC = 1;
+    if (EnGo_IsEnteringSleep(this, play)) {
+        this->sleepState = ENGO_ASLEEP_POS;
     }
 
-    if (this->unk_390 & 0x4000) {
+    if (this->actionFlags & ENGO_FLAG_ROLLINGUP) {
```

Comment:
It occurs to me that usually we say Gorons curl up, not roll up.

Reply:
Originally had that, but switched to "Roll" after noticing z_en_s_goro uses that terminology. Preferred to keep it the same as the prior code. Could probably update both if it's a strong preference.